libs/typescript/cosmos-utils/testnet-setup.ts

```diff
@@ -2,24 +2,21 @@ import { exec as exec_, execSync } from 'node:child_process';
 import { promisify } from 'node:util';
 
 import { SigningCosmWasmClient } from '@cosmjs/cosmwasm-stargate';
-import { DirectSecp256k1HdWallet, OfflineSigner } from '@cosmjs/proto-signing';
+import { DirectSecp256k1HdWallet } from '@cosmjs/proto-signing';
+
+import { getRootDir, sleep } from '../ts-utils/common-utils';
+import { cosmosUtils, cosmosWalletInfo } from './cosmos-utils';
 
 const exec = promisify(exec_);
-function sleep(ms) {
-  return new Promise(resolve => setTimeout(resolve, ms));
-}
 
 export async function setUpCosmosTestnet(
-  rootDir: string,
   rpcEndpoint: string,
   signal: AbortSignal,
 ) {
-  let DendrETHWalletInfo = {
-    mnemonic:
-      'economy stock theory fatal elder harbor betray wasp final emotion task crumble siren bottom lizard educate guess current outdoor pair theory focus wife stone',
-    address: '',
-  };
-
+  let DendrETHWalletInfo = new cosmosWalletInfo(
+    String(process.env['COSMOS_LOCAL_MNEMONIC']),
+  );
+  const rootDir = await getRootDir();
```

Comment:
Consider moving this function to bash:
1. Most of what it does is call other bash scripts
2. The `getFredAddressCommand` should not be necessary as we can derive the private keys from the mnemonic and we can derive the public key from the private key.

I think in the end we only need a function along the lines of:
```ts
export startLocalCosmosNode(mnemonic: string): { rpcUrl: string };

export function getCosmosTxClient(mnemonic: string, network: 'wasm' | 'cudos', rpcUrl: string): CosmosClinetWithWallet {
  const wallet = await DirectSecp256k1HdWallet.fromMnemonic(
    mnemonic,
    {
      prefix: network,
    },
  );
  const client = await SigningCosmWasmClient.connectWithSigner(
    rpcEndpoint,
    wallet,
  );
  return { client, wallet };
}

// in a galaxy far away...

const { rpcUrl } = await startLocalCosmosNode(yargs.mnemonic, 'wasm');
const clientWithWallet = await getCosmosTxClient(yargs.mnemonic, rpcUrl);
```